rpc/namenode.go

```diff
@@ -309,27 +321,49 @@ func (c *NamenodeConnection) readResponse(method string, resp proto.Message) err
 // |  RPC service class, 1 byte (0x00)                         |
 // +-----------------------------------------------------------+
 // |  Auth protocol, 1 byte (Auth method None = 0x00)          |
+// +-----------------------------------------------------------+
+
+// +-----------------------------------------------------------+
+// | Kerberos Auth handshake happens here, if required.        |
+// +-----------------------------------------------------------+
+
 // +-----------------------------------------------------------+
 // |  uint32 length of the next two parts                      |
 // +-----------------------------------------------------------+
 // |  varint length + RpcRequestHeaderProto                    |
 // +-----------------------------------------------------------+
 // |  varint length + IpcConnectionContextProto                |
 // +-----------------------------------------------------------+
+
 func (c *NamenodeConnection) writeNamenodeHandshake() error {
+
+	var authProto byte = noneAuthProtocol
+	if c.kerberosClient != nil {
+		authProto = saslAuthProtocol
+	}
+
 	rpcHeader := []byte{
 		0x68, 0x72, 0x70, 0x63, // "hrpc"
-		rpcVersion, serviceClass, authProtocol,
+		rpcVersion,
+		serviceClass,
+		authProto,
+	}
+
+	c.conn.Write(rpcHeader)
+
+	if authProto == saslAuthProtocol {
+		doKerberosHandshake(c)
 	}
```

Comment:
Kindly handle error condition here. Even in case of failed kerberos authentication, it doesn't return here and leads to `FATAL_INVALID_RPC_HEADER`. Kindly fix the issue.

Reply:
Thanks, missed that one. It's obviously a mistake :)